Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -0,0 +1,1439 @@
+    ///   - messageId: The id of the message to remove the reaction from.
+    ///   - type: The type that describes a message reaction. Common examples are: “like”, “love”, “smile”, etc. An user can have only 1 reaction of each type per message.
+    public func deleteReaction(
+        from messageId: MessageId,
+        with type: MessageReactionType
+    ) async throws {
+        try await messageUpdater.deleteReaction(type, messageId: messageId)
+    }
+    
+    /// Adds a reaction to the specified message.
+    ///
+    /// - Note: Custom data for reactions is limited to 1 KB.
+    ///
+    /// - Parameters:
+    ///   - messageId: The id of the message to send the reaction.
+    ///   - type: The type that describes a message reaction. Common examples are: “like”, “love”, “smile”, etc. An user can have only 1 reaction of each type per message.
+    ///   - score: The score of the reaction for cumulative reactions (example: n number of claps).
+    ///   - enforceUnique: If `true`, the added reaction will replace all reactions the user has (if any) on this message.
+    ///   - extraData: The reaction's extra data.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    public func sendReaction(
+        to messageId: MessageId,
+        with type: MessageReactionType,
+        score: Int = 1,
+        enforceUnique: Bool = false,
+        extraData: [String: RawJSON] = [:]
+    ) async throws {
+        try await messageUpdater.addReaction(
+            type,
+            score: score,
+            enforceUnique: enforceUnique,
+            extraData: extraData,
+            messageId: messageId
+        )
+    }
+    
+    /// Loads reactions for the specified message and pagination parameters.
+    ///
+    /// All the currently loaded reactions can be accessed through ``MessageState/reactions``.
+    ///
+    /// - Parameters:
+    ///   - messageId: The id of the message to load reactions.
+    ///   - pagination: The pagination configuration which includes a limit and an offset or a cursor.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    /// - Returns: An array of reactions for given limit and offset.
+    @discardableResult public func loadReactions(
+        for messageId: MessageId,
+        pagination: Pagination
+    ) async throws -> [ChatMessageReaction] {
+        try await messageUpdater.loadReactions(
+            cid: cid,
+            messageId: messageId,
+            pagination: pagination
+        )
+    }
+    
+    /// Loads more reactions and updates ``MessageState/reactions``.
+    ///
+    /// - Parameters:
+    ///   - messageId: The id of the message to load reactions.
+    ///   - limit: The limit for the page size. The default limit is 25.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    /// - Returns: An array of reactions for the next page.
+    @discardableResult public func loadMoreReactions(
```

Comment:
It kind of depends. It's always a tradeoff between ease of use for customers and maintenance. When I think about how customers would use this, I can imagine more people needing to load members from the chat, so that one should stay. About this one, I'm not entirely sure.

Reply:
Discuss on tech discussion, for now we will keep it like this and have both ways